pkg/scheduler/framework/interface.go

```diff
@@ -48,6 +49,12 @@ type PluginToNodeScores map[string]NodeScoreList
 // NodeToStatusMap declares map from node name to its status.
 type NodeToStatusMap map[string]*Status
 
+// Diagnosis records the details to diagnose a scheduling failure.
+type Diagnosis struct {
+	NodeToStatusMap NodeToStatusMap
+	FailedPlugins   sets.String
```

Comment:
where is this Set being used? I see it being populated, but not used.

Reply:
True, it will be used in the following PR(s).

Upon recordingFailure(), if the error is of type FitError, the FailedPlugins will be injected into PodInfo. Then upon moving pods bwteeen internal queues, we have (1) the triggered cluster events, (2) registered map from event to plugins, and (3) failedPlugins carried by PodInfo. So we're able to tell whether to move a Pod(Info) or not.

Reply:
ah, but once policy is deprecated (and so the runAllPlugins flag), then we don't really need this because the FailedPlugin in status should suffice.

Reply:
But then again, runAllPlugins is disabled by default and will eventually get deprecated. I think failedPlugin in status will probably be enough (in the worst case, if runAllPlugins is enabled, then moving pods back to the queue will simply be a little less efficient).

Reply:
Oh, you may misunderstand this: this FaieldPlugins actually is deduced from `NodeToStatusMap` like this:

```golang
for _, status := range nodeToStatusMap
    failedPlugins.Insert(status.FailedPlugin())
```

So its length equals to length of all nodes. No matter runAllPlugins is enabled or not, a single Fitler plugin return one status (for that node).

BTW: another thing I want to mentions is: given we always process one cluster event at a time, so if a Pod failed by multiple plugins (e.g., foo and bar), it's qualified to be moved once an event associated with foo **or** bar is triggered.

Reply:
Ok, sounds good, sorry I misread the logic.